elastic/kibana
# [Infrastructure UI] Add link to k8s dashboard and refactor Inventory UI

## Summary

closes [#149626](https://github.com/elastic/kibana/issues/149616)

This PR adds a new experimental link to the Inventory UI, that will redirect the user to the list of managed Kubernetes Dashboards.

- Before the user interacts with the link
<img width="829" alt="image" src="https://user-images.githubusercontent.com/2767137/216944804-9b6b7462-feef-4770-b49d-86d48eef2dc1.png">

<img width="811" alt="image" src="https://user-images.githubusercontent.com/2767137/216944862-aa4f997c-5736-4c08-a0d2-a235df5fab56.png">


- After the user interacts with the link
<img width="1359" alt="image" src="https://user-images.githubusercontent.com/2767137/216944639-27ad8c07-d9ba-4217-89a1-855583e2aa76.png">


### How to test it
pre: Make sure you have `Kubernetes` integration package installed 

- Go to `Observability > Inventory`
  - If the button hasn't been clicked yet, the link should be pink and the badge should be present
- Click on the badge and/or the link
- The list of managed Kubernetes dashboards will render (if you haven't installed the integration package, the list will be empty)
- Return to the Inventory UI
  - The link color will be blue and the badge will be hidden

We populate a localStorage key named `inventoryUI:k8sDashboardClicked`

### For maintainers

- I've also refactored the Inventory UI `layout.tsx` file, removing some usages of the `AutoSizer` component in favor of css.
